lib/mongo/client.rb

```diff
@@ -902,9 +904,14 @@ def watch(pipeline = [], options = {})
 
     # Create a new encrypter object using the client's auto encryption options
     def build_encrypter
-      @encrypter = Crypt::AutoEncrypter.new(
-        @options[:auto_encryption_options].merge(client: self)
-      )
+      begin
+        @encrypter = Crypt::AutoEncrypter.new(
+          @options[:auto_encryption_options].merge(client: self)
+        )
+      rescue => e
+        do_close
```

Comment:
I am confused as to what this call is intended to do:

1. It also disconnects the cluster, why does a failure to create encrypter cause the cluster to be disconnected?
2. I don't think it would do anything as far as closing the encrypter if `@encrypter` is not assigned, therefore, if there's an exception raised in the block above this call wouldn't actually perform any encrypter cleanup?

What I think should happen is the `AutoEncrypter` constructor needs to have the begin-rescue block in it around `EncryptionIO` instantiation (or more precisely everything after the `Client` is constructed there) and if any exception is raised, the client that was made should be closed.

Reply:
We talked over Slack and agreed that:
- The client should disconnect the cluster here
- There should also be a begin... rescue block in the Encrypter initializer that closes the mongocryptd client if encryption fails